hack/ensure-go.sh

```diff
@@ -31,7 +31,7 @@ EOF
   local go_version
   IFS=" " read -ra go_version <<< "$(go version)"
   local minimum_go_version
-  minimum_go_version=go1.20.0
+  minimum_go_version=go1.21.6
```

Comment:
Not something we have to address now, but I'm wondering if we need this script at all since I think the `go` directive in go.mod already enforces this everywhere we'd need. I imagine this script was probably added to k/k like 8 years ago before Go modules and we inherited it from there.